lib/request.js

```diff
@@ -85,6 +85,11 @@ function buildRequestWithTrustProxy (R, trustProxy) {
 }
 
 Object.defineProperties(Request.prototype, {
+  id: {
+    get () {
+      return this._id
+    }
+  },
```

Comment:
This seems totally unnecessary to me, can you restore it the way it was?

Reply:
> Also encapsulated id property as it's not meant to be mutated
This is just for this purpose. Otherwise we can do:
```javascript

Object.defineProperty(this, 'id', {
  value: id,
  writable: false
});
```

What do you think?

Reply:
There is no need for this. It's feasible to mutate all these properties anyway, so let's not make things slower for no actual gain.

Reply:
will fix

Reply:
That's the reason that makes me not sure about flagging these props as readonly in TS. I think that stick to JS is the best we can do. :)